pallets/micropayment/src/lib.rs

```diff
@@ -529,20 +527,13 @@ pub mod pallet {
 
         /// Deduct the amount from the account free balance
         fn take_from_account(account: &T::AccountId, amount: BalanceOf<T>) -> bool {
-            let result = T::Currency::mutate_account_balance(account, |balance| {
-                let min_balance = T::Currency::minimum_balance();
-                // ensure after taking amount from account, remaining balance is grater than min_balance
-                if balance.free <= amount.saturating_add(min_balance) {
-                    return Zero::zero();
-                } else {
-                    balance.free -= amount;
-                }
-                return amount;
-            });
-            match result {
-                Ok(actual_amount) => actual_amount == amount,
-                _ => false,
-            }
+            T::Currency::withdraw(
```

Comment:
https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L907-L909
`withdraw` and `deposit_creating` use `try_mutate_account`, and they will decrease and increase the total issuance, looks like not very good.
In this situation I think using a lock may be more appropriate, like the https://docs.substrate.io/rustdocs/latest/pallet_vesting/index.html vesting pallet.
When a user start using micropayment channel, we some some balance, when a server `claim_payment`, he just get from the locked balance.
It's a little off-topic, but we can discuss whether to change in the future.

Reply:
Withdrawing ，when claming payment ,then deposit the value.  The total issurance  unchanged.
But there is temporary unbalanced, maybe we should withdraw the balance to micropayment pallet-id account, then pay fee from pallet-id account